src/itemstackmetadata.cpp

```diff
@@ -40,4 +56,29 @@ void ItemStackMetadata::deSerialize(std::istream &is)
 		// BACKWARDS COMPATIBILITY
 		m_stringvars[""] = in;
 	}
+	updateToolCapabilities();
+}
+
+void ItemStackMetadata::updateToolCapabilities()
+{
+	if (contains(TOOLCAP_KEY)) {
+		toolcaps_overridden = true;
+		toolcaps_override = ToolCapabilities();
+		std::istringstream is(getString(TOOLCAP_KEY));
+		toolcaps_override.deserializeJson(is);
```

Comment:
maybe add a comment like: 

> Will fails on the empty string, giving just ToolCapabilities()

Reply:
If it's empty string, then wouldn't TOOLCAP_KEY not be contained?

Reply:
`Metadata::setString` removes the field, so it seems to be okay.

Reply:
this is needed desperately for a tinkers construct clone I'm working on